var/spack/repos/builtin/packages/foam-extend/package.py

```diff
@@ -0,0 +1,265 @@
+    variant('scotch', default=True, description='Activate Scotch as a possible decomposition library')
+    variant('ptscotch', default=True, description='Activate PT-Scotch as a possible decomposition library')
+    variant('metis', default=True, description='Activate Metis as a possible decomposition library')
+    variant('parmetis', default=True, description='Activate Parmetis as a possible decomposition library')
+    variant('parmgridgen', default=True, description='Activate Parmgridgen support')
+
+    supported_compilers = {'clang': 'Clang', 'gcc': 'Gcc', 'intel': 'Icc'}
+
+    depends_on('mpi')
+    depends_on('cmake')
+    depends_on('python')
+    depends_on('flex@:2.5.99')
+    depends_on('bison')
+    depends_on('m4')
+    depends_on('hwloc')
+    depends_on('zlib')
+
+    depends_on('scotch ~ metis', when='~ptscotch+scotch')
+    depends_on('scotch ~ metis + mpi', when='+ptscotch')
+    depends_on('metis@5:', when='+metis')
+    depends_on('parmetis', when='+parmetis')
+    depends_on('parmgridgen', when='+parmgridgen')
+
+    depends_on('paraview', when='+paraview')
+    depends_on('qt@:4', when='+paraview')
+
+    def _get_env(self, command):
+        variable_set = re.compile(r'(.+?)=(.*)')
+        proc = subprocess.Popen(['bash', '-c', command],
+                                stdout=subprocess.PIPE)
+        env = {}
+        for line in proc.stdout:
+            match_grp = variable_set.match(line)
+            if match_grp:
+                env[match_grp.group(1)] = match_grp.group(2)
+        return env
+
+    def set_arch(self):
+        (sysname, nodename, release, version, machine) = os.uname()
+
+        if self.compiler.name not in self.supported_compilers:
+            raise RuntimeError('{0} is not a supported compiler \
+            to compile OpenFOAM'.format(self.compiler.name))
+
+        foam_compiler = self.supported_compilers[self.compiler.name]
+        if sysname == 'Linux':
+            arch = 'linux'
+            if foam_compiler == 'Clang':
+                raise RuntimeError('OS, compiler combinaison not\
+                supported ({0} {1})'.format(sysname, foam_compiler))
+        elif sysname == 'Darwin':
+            if machine == 'x86_64':
+                arch = 'darwinIntel'
+            if foam_compiler == 'Icc':
+                raise RuntimeError('OS, compiler combinaison not\
+                supported ({0} {1})'.format(sysname, foam_compiler))
+        else:
+            raise RuntimeError('{0} {1} is not a \
+            supported architecture'.format(sysname, machine))
+
+        return (arch, foam_compiler)
+
+    def get_openfoam_environment(self, env_openfoam):
+        env_before = self._get_env('env')
+
+        with working_dir(self.stage.source_path):
+            env_after = self._get_env('source etc/bashrc && env')
+
+        if 'PATH' in env_after:
+            del env_after['PATH']
+
+        if 'LD_LIBRARY_PATH' in env_after:
+            del env_after['LD_LIBRARY_PATH']
```

Comment:
In fact as @adamjstewart said to use or compile *foam* you have to source this `etc/bashrc`

And it is searched by some tools in a folder like `$FOAM_INST_DIR/foam-{version}/etc`
This raises a question, in the module I just added this FOAM_INST_DIR variable but it would not be difficult to add all the variables set by this `bashrc` file.
 It would make it more convenient for the users, but add lots of variables in a module.
Another solution would be to have a variable containing the full path to this file. To simplify the sourcing step.